sources/engine/Stride.UI/Renderers/DefaultTextBlockRenderer.cs

```diff
@@ -47,6 +47,38 @@ public override void RenderColor(UIElement element, UIRenderingContext context)
                 Batch.BeginCustom(context.GraphicsContext, 1);                
             }
 
+            
+
+            if (textBlock.OutlineColor != null && textBlock.OutlineThickness > 0)
+            {
+                var borderThickness = textBlock.OutlineThickness;
+                var borderColor = textBlock.RenderOpacity * textBlock.OutlineColor;
+
+                
+                Vector2[] offsets =
```

Comment:
This array could be stackalloc'ated instead or replace the loop below with a loop going from`x -1 -> +1` and an inner one `y -1 -> +1`.
Improves perfs and reduce GC hit.

Reply:
Thanks for the suggestion I will try to work on that tomorrow

Reply:
so i changed the code to 

- use nested loops instead of a Vector2 array 
- the code a lot cleaner and easier to read
- Old: Pre-calculated array of positions
- New: Calculated on-the-fly using nested loops

Reply:
it's true that changing the **font shader** would be **more accurate and i guess faster** my current knowledge base still didn't reach that level but i am improving i might work on it on future but this feature could be as a placeholder until i or someone else implement a better solution i am learning more about ui rendering and i might practice sometimes or even contribute to stride ui since it does need some features